package/com.unity.formats.usd/Editor/Recorder/UsdRecorder.cs

```diff
@@ -0,0 +1,104 @@
+#if RECORDER_AVAILABLE
+using System.IO;
+using pxr;
+using Unity.Formats.USD;
+using UnityEditor.Recorder;
+using UnityEngine;
+
+namespace UnityEditor.Formats.USD.Recorder
+{
+    public class UsdRecorder : GenericRecorder<UsdRecorderSettings>
+    {
+        ExportContext context;
+        UsdRecorderInput Input => m_Inputs[0] as UsdRecorderInput;
+
+        // Stateful stuff used during the recording that should be cleaned
+        DirectoryInfo usdzTemporaryDir;
+        string usdcFileName;
+        string usdzFileName;
+        string usdzFilePath;
+        string currentDir;
+        protected override void SessionCreated(RecordingSession session)
+        {
+            base.SessionCreated(session);
+
+            InitUsd.Initialize();
+            var outputFile = Settings.FileNameGenerator.BuildAbsolutePath(session);
+            if (Settings.ExportFormat == UsdRecorderSettings.Format.UsdZ) // FIXME Support USDz
+            {
+                currentDir = Directory.GetCurrentDirectory();
+                // Setup a temporary directory to export the wanted USD file and zip it.
+                var tmpDirPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+                usdzTemporaryDir = Directory.CreateDirectory(tmpDirPath);
+
+                // Get the usd file name to export and the usdz file name of the archive.
+                usdcFileName = Path.GetFileNameWithoutExtension(outputFile) + ".usdc";
+                usdzFileName = Path.GetFileName(outputFile);
+                var fi = new FileInfo(outputFile);
+                usdzFilePath = fi.FullName;
+
+                // Set the current working directory to the tmp directory to export with relative paths.
+                Directory.SetCurrentDirectory(tmpDirPath);
+
+                outputFile = Path.Combine(tmpDirPath,usdcFileName);
+            }
+
+            context = new ExportContext
+            {
+                scene = ExportHelpers.InitForSave(outputFile)
```

Comment:
Will leave it be for now: the error message is not too shabby and because of the tokens in the name this be a pre-validation.
If this occurs, Recording stops, Exists playmode and a console and recorder window error message is logged:
"ApplicationException: USD ERROR: A layer already exists with identifier '/d-unity-sdk/TestProject/Usd-Recorder/Recordings/Usd Clip.usd'"